### PR TITLE
In the Sorting Widget, use the title for the vocabulary term rather than the index title.

### PR DESCRIPTION
**Pull request by tsimkins**
# Pull request notes

This is a small change that I patched into our sites, and I wanted to pass it upstream.  I thought this was a more logical behavior for the widget, but it will change the displayed value in the dropdown for the sorting widget depending on what users have configured in the vocabulary.

Feel free to reject if you think it will confuse users that already use a vocabulary for the sorting widget.  However, my guess is that if someone is savvy enough to create vocabulary, this won't be a problem, and would be a helpful feature.

# Change summary

The Sorting widget has an option for a "Vocabulary to use to filter sorting criteria".  However, that displays the index title from the catalog (e.g. "Sortable Title") and ignores the title of the vocabulary term.

This change will use the title for the vocabulary term rather than the "friendly" index title.

# Use case

My use case for this was that I wanted to present Sorting widget dropdown, but limit the fields to these indexes:

  * `sortable_title`
  * `effective`
  * `created`

which can be filtered by a vocabulary.  

However, the field titles are the same "friendly" values used by collections, which are generic.  For example, `sortable_title` is simply "Sortable Title".

Although these can be set in the registry (via the `plone.app.querystring.field.sortable_title.title` key), I only wanted to set the title in the widget, not site-wide.

In other words, "Sortable Title" or "Title" might not be intuitive to the user if we had a faceted nav on a group of people.  For that page (only), we might want to use "Sort by ... Person Name", since "Title" wouldn't make sense, even though the person's name was the `.Title()` of the object.

Being able to use a vocabulary title per-widget seemed like a logical place to set that.

For the other items (`effective` and `created`), we may want to use different terms that are context-specific, rather than CMS-specific.

### CODE REVIEW

**Comment by avoinea** [comment]
@tsimkins Interesting... Thank you ;)

File: eea/facetednavigation/widgets/sorting/widget.py [hunk 1]
```diff
@@ -109,9 +109,12 @@ def vocabulary(self, **kwargs):
...
         if not vocab:
-            return [field for field in sort_fields]
+            return sort_fields
+
+        vocab_fields = [(x[0], x[1], '') for x in vocab]
+        sort_field_ids = [x[0] for x in sort_fields]
 
-        vocab_fields = [field[0].replace('term.', '', 1) for field in vocab]
-        return [f for f in sort_fields if f[0] in vocab_fields]
+        return [f for f in vocab_fields if f[0] in sort_field_ids]
```
Comment (avoinea) [inline eea/facetednavigation/widgets/sorting/widget.py]: Ops. You removed `field[0].replace('term.', '', 1)`. I think this is helpful when you use it with Products.ATVocabularyManager vocabularies. 

@tsimkins Can you fix this, please?
